.github/workflows/check.yaml

```diff
@@ -14,6 +20,10 @@ concurrency:
   group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
   cancel-in-progress: true
 
+env:
+  # TEST_JUJU3 needed due https://github.com/openstack-charmers/zaza/blob/b22c2eed4c322f1dfc14ffb2d31e0dd18c911a40/setup.py#L47 to support Juju3+
+  func_cmd: TEST_JUJU3=1 make functional ${{ github.event.inputs.debug_enabled && "--keep-faulty-model" }}
```

Comment:
How this supposes to work? If debug_enable is true  the `TEST_JUJU3=1 make functional true --keep-faulty-model` will be executed?

I think we should use FUNC_ARGS to pass arguments to functional tests, because this approach will not work. Is this even a wise approach? Who will clean up the model? I think we should use "--keep-faulty-model" by default in Makefile, same as we do in tox and simply add juju crushdump and clean up process. WDYT?

Reply:
I think we'll always need some way to be able to interactively debug a failing test (think something like tmate), and at the moment the model is torn down too quickly for me to log in and understand why it's failing. I'm not sure how I would do that with only a juju crashdump. Is the concern that we would forget to destroy the model? If that's the issue maybe we can simply have the debug switch extend the timeout to a few hours

Reply:
Using [tmate](https://github.com/mxschmitt/action-tmate) would be a better option. 

> Is the concern that we would forget to destroy the model? 

Yes, I remember how many models we have on serverstack-ci by Jenkins. 

IMO, we should collect juju-crashdump always, even if tests do not failed and then destroy model. With logs, I can verify that tests are actually doing what I'm expecting. 

1. Create a model
2. Run tests on specific model
3. show juju status, microk8s status, lxd status, ... enything from env
4. collect logs (with juju-crushdump)
5. enable tmate if debug is true
6. destroy-model

The action-operator will clean up for us and for GH runners we even do not need to clean up. By creating a model, we can simply run juju status, crashdump against model name without need to find the name.